engine/CMakeLists.txt

```diff
@@ -1,29 +1,30 @@
-add_subdirectory(audio)
+set(TARGET_NAME milg-engine)
+set(
+    SOURCE_FILES
+    "src/core/application.cpp"
+    "src/core/asset.cpp"
+    "src/core/asset_store.cpp"
+    "src/core/imgui_layer.cpp"
+    "src/core/layer.cpp"
+    "src/core/logging.cpp"
+    "src/core/window.cpp"
 
-set(TARGET_NAME milg)
+    "src/audio/engine.cpp"
+    "include/milg/audio/engine.hpp"
+    "include/milg/audio/node.hpp"
+    "src/audio/sound.cpp"
+    "include/milg/audio/sound.hpp"
+    "src/audio/vocoder_node.cpp"
+    "include/milg/audio/vocoder_node.hpp"
 
-set(
-    GRAPHICS_SOURCE_FILES
-    "graphics/buffer.cpp"
-    "graphics/swapchain.cpp"
-    "graphics/texture.cpp"
-    "graphics/vk_context.cpp"
-    "graphics/pipeline.cpp"
-    "graphics/sprite_batch.cpp"
-)
+    "src/graphics/buffer.cpp"
+    "src/graphics/swapchain.cpp"
+    "src/graphics/texture.cpp"
+    "src/graphics/vk_context.cpp"
+    "src/graphics/pipeline.cpp"
+    "src/graphics/sprite_batch.cpp"
+    "src/graphics/map.cpp"
```

Comment:
Fwiw, you can just pass the paths to `add_library()`, no need for variables.

Reply:
yeah, i default to using a variable, since it becomes trivial to add to the list based on configuration, if we need (or don't need) to compile some features in the future